nmap_dns.cc

```diff
@@ -599,7 +616,11 @@ static int deal_with_timedout_reads() {
             output_summary();
             stat_dropped++;
             total_reqs--;
-            delete tpreq;
+            infoI = records.find(tpreq->id);
+            if ( infoI != records.end() ){
```

Comment:
This condition show our unconfidence in presence of record in map, I think we can remove this. I tested without this condition also and it works fine.